### PR TITLE
Pulling change entity rotation

### PR DESCRIPTION
**Pull request by Macoron**
## About the PR 
Now when you pull some object with Rotatable component, it will change object rotation. Doesn't work on mobs.
Moved RotatableComponent to Shared namespace.

**Screenshots**

https://user-images.githubusercontent.com/6161335/115990706-4d60cb80-a5cd-11eb-8a4f-9bd27916b212.mp4

**Changelog**

:cl:
- tweak: Chairs and other object will change their direction when you pull them.

### CODE REVIEW

File: Content.Shared/GameObjects/EntitySystems/SharedPullingSystem.cs [hunk 1]
```diff
@@ -198,5 +218,25 @@ public bool IsPulling(IEntity puller)
         {
             return _pullers.ContainsKey(puller);
         }
+
+        private void UpdatePulledRotation(IEntity puller, IEntity pulled)
+        {
+            if (!pulled.TryGetComponent(out SharedRotatableComponent? rotatable))
+                return;
```
Comment (metalgearsloth) [inline Content.Shared/GameObjects/EntitySystems/SharedPullingSystem.cs:225]: Slap a TODO on this to update once ComponentReference works with directed event bus.

File: Content.Shared/GameObjects/EntitySystems/SharedPullingSystem.cs [hunk 2]
```diff
@@ -198,5 +218,25 @@ public bool IsPulling(IEntity puller)
...
+
+            var dir = puller.Transform.WorldPosition - pulled.Transform.WorldPosition;
+            if (dir.LengthSquared > ThresholdRotDistance * ThresholdRotDistance)
+            {
+                var oldAngle = pulled.Transform.LocalRotation;
+                var newAngle = Angle.FromWorldVec(dir);
+
+                var diff = newAngle - oldAngle;
+                if (Math.Abs(diff.Degrees) > ThresholdRotAngle)
+                    pulled.Transform.LocalRotation = newAngle;
```
Comment (metalgearsloth) [inline Content.Shared/GameObjects/EntitySystems/SharedPullingSystem.cs:238]: Should this be comparing local angles with world angles? Won't it look weird with rotated shuttles?

**Comment by Macoron** [comment]
Fixed

File: Content.Shared/GameObjects/EntitySystems/SharedPullingSystem.cs [hunk 3]
```diff
@@ -198,5 +218,26 @@ public bool IsPulling(IEntity puller)
...
+
+            var dir = puller.Transform.WorldPosition - pulled.Transform.WorldPosition;
+            if (dir.LengthSquared > ThresholdRotDistance * ThresholdRotDistance)
+            {
+                var oldAngle = pulled.Transform.WorldRotation;
+                var newAngle = Angle.FromWorldVec(dir);
+
+                var diff = newAngle - oldAngle;
+                if (Math.Abs(diff.Degrees) > ThresholdRotAngle)
+                    pulled.Transform.LocalRotation = newAngle;
```
Comment (metalgearsloth) [inline Content.Shared/GameObjects/EntitySystems/SharedPullingSystem.cs:239]:
```suggestion
                    pulled.Transform.WorldRotation = newAngle;
```

**Pull request merged**